app/controllers/pages_controller.rb

```diff
@@ -1,4 +1,6 @@
 class PagesController < ApplicationController
   def index
+    return if !params[:city_name] || Location.all.any? { |a| a.city_name == params[:city_name] }
+    raise RoutingError, 'Not Found'
```

Comment:
Stay positive. Generally speaking, prefer positive conditions over inverting conditions or reversing method return short circuits e.g.
`raise RoutingError, 'Not Found' if params[:city_name].empty? || ...`

In this case, I think we should probably hack in a `redirect_to ...` on empty param to the city_name of our PRIMARY_LOCATION to remove the need to handle primary location in our javascript code.